hal/src/dma.rs

```diff
@@ -0,0 +1,3 @@
+//! Direct Memory Access
+
+pub mod controller;
```

Comment:
This module is empty right now, but the goal is to put the DMA driver in here. However, it needs some way of accessing the DMAC registers first.